src/components/FlexibleFee/FlexibleFee.tsx

```diff
@@ -0,0 +1,276 @@
+import { useTokenBalance } from 'src/hooks/useTokenBalance';
+
+interface FlexibleFeeProps {
+  route: RouteExtended;
+}
+
+const MIN_AMOUNT = 1.5;
+
+// TODO:
+
+export const FlexibleFee: FC<{ route: RouteExtended }> = ({
+  route,
+}: FlexibleFeeProps) => {
+  console.log('FlexibleFee-ROUTE....', route);
+  const [activeChain, setActiveChain] = useState<ExtendedChain | undefined>(
+    undefined,
+  );
+  const [balance, setBalance] = useState<number>(0);
+  const [balanceUSD, setBalanceUSD] = useState<number>(0);
+  const [amount, setAmount] = useState<string>('0');
+  const [rate, setRate] = useState<string>('0.3');
+  const { accounts } = useAccounts();
+  const activeAccount = accounts.filter((account) => account.isConnected);
+  const theme = useTheme();
+  const ref = useRef<HTMLInputElement>(null);
+  const { chains } = useChains();
+  const { t } = useTranslation();
+  const { switchChainAsync } = useSwitchChain();
+  const {
+    data: transactionHash,
+    isPending,
+    sendTransaction,
+  } = useSendTransaction();
+  const { isLoading: isConfirming, isSuccess: isConfirmed } =
+    useWaitForTransactionReceipt({
+      hash: transactionHash,
+    });
+
+  const { data: sourceBalance, isLoading: isSourceBalanceLoading } =
+    useTokenBalance({
+      tokenAddress: '0x0000000000000000000000000000000000000000',
+      walletAddress: activeAccount[0]?.address as `0x${string}`,
+      chainId: route.fromChainId,
+    });
+
+  const { data: destinationBalance, isLoading: isDestinationBalanceLoading } =
+    useTokenBalance({
+      tokenAddress: '0x0000000000000000000000000000000000000000',
+      walletAddress: activeAccount[0]?.address as `0x${string}`,
+      chainId: route.toChainId,
+    });
+
+  // const { data: sourceBalance, isLoading: isSourceBalanceLoading } = useBalance(
+  //   {
+  //     address: activeAccount[0]?.address as `0x${string}`,
+  //     chainId: route.fromChainId,
+  //   },
+  // );
+  // const { data: destinationBalance, isLoading: isDestinationBalanceLoading } =
+  //   useBalance({
+  //     address: activeAccount[0]?.address as `0x${string}`,
+  //     chainId: route.toChainId,
+  //   });
+
+  useEffect(() => {
+    console.log('destinationBalance', destinationBalance);
+    console.log('sourceBalance', sourceBalance);
+    if (
+      sourceBalance?.amount &&
+      (Number(sourceBalance.amount) / 10 ** sourceBalance.decimals) *
+        parseFloat(sourceBalance.priceUSD) >=
+        MIN_AMOUNT
+    ) {
+      setBalance(Number(sourceBalance.amount) / 10 ** sourceBalance.decimals);
+      setActiveChain(
+        chains?.find((chainEl) => chainEl.id === route.fromChainId),
+      );
+    } else if (
+      destinationBalance?.amount &&
+      (Number(destinationBalance.amount) / 10 ** destinationBalance.decimals) *
+        parseFloat(destinationBalance.priceUSD) >=
+        MIN_AMOUNT
+    ) {
+      setBalance(
+        Number(destinationBalance.amount) / 10 ** destinationBalance.decimals,
```

Comment:
Please use viem format utils for conversions 

https://viem.sh/docs/utilities/formatUnits
https://viem.sh/docs/utilities/parseUnits